consul/server_manager/server_manager.go

```diff
@@ -133,9 +154,38 @@ func (sc *serverConfig) cycleServer() (servers []*server_details.ServerDetails)
 	newServers := make([]*server_details.ServerDetails, 0, numServers)
 	newServers = append(newServers, sc.servers[1:]...)
 	newServers = append(newServers, sc.servers[0])
+
+	// FIXME(sean@): Is it worth it to fire off a go routine and
+	// PingConsulServer?
 	return newServers
 }
 
+// removeServerByKey performs an inline removal of the first matching server
+func (sc *serverConfig) removeServerByKey(targetKey *server_details.Key) {
+	for i, s := range sc.servers {
+		if targetKey.Equal(s.Key()) {
+			// Delete the target server
+			copy(sc.servers[i:], sc.servers[i+1:])
+			sc.servers[len(sc.servers)-1] = nil
+			sc.servers = sc.servers[:len(sc.servers)-1]
+			return
+		}
+	}
+}
+
+// shuffleServers shuffles the server list in place
+func (sc *serverConfig) shuffleServers() {
+	newServers := make([]*server_details.ServerDetails, len(sc.servers))
```

Comment:
There was a revision in the past where this shuffle could be rolled back, however that is no longer necessary.